File: ros_ws/src/intro_pkg1/src/ServiceClient1.cpp
// Header files for ROS
#include "ros/ros.h"
// For service
#include "intro_pkg1/FloatIO.h"
// For atof
#include "stdlib.h"

int main(int argc, char **argv) {
    // Initialize the node
    ros::init(argc, argv, "FloatIO_ServiceClient");
    // Create a node handler
    ros::NodeHandle nodeHandler;
    // Create a Service Client
    ros::ServiceClient client = nodeHandler.serviceClient<intro_pkg1::FloatIO>("equation_service");
    // Service object
    intro_pkg1::FloatIO srv;
    // Assign value to request
    srv.request.input = atof(argv[1]);
    // Call the service
    if (client.call(srv)) {
        // srv.response now has value
        ROS_INFO("X = %f, Y = %f", srv.request.input, srv.response.output);
    } else {
        ROS_ERROR("Service call failed");
    }
    return 0;
}

File: ros_ws/src/src/intro_pkg1/scripts/ServiceClient1.py
#!/usr/bin/env python
# Import libraries
import rospy
# Service files
from intro_pkg1.srv import *
# System files
import sys

def main():
    # Initialize the node
    rospy.init_node("FloatIO_ServiceClient", anonymous=True)
    # Initialize Service Client
    serviceClient = rospy.ServiceProxy("equation_service", FloatIO)
    # Call the service and then receive response
    response = serviceClient(float(sys.argv[1]))
    rospy.loginfo("X = {0}, Y = {1}".format(float(sys.argv[1]), response))

if __name__ == "__main__":
    try:
        main()
    except rospy.ROSInterruptException:
        rospy.logfatal("Node crashed")
